client/client.go

```diff
@@ -46,3 +49,34 @@ func (c *Client) newRequest(method, url string, body io.Reader) (*http.Request,
 	}
 	return req, nil
 }
+
+// ServerMetadata represents the server metadata response
+type ServerMetadata struct {
```

Comment:
This type definition should go inside the `pkg/types` package, it contains shared data types.
Then this struct can be shared by both the client and the server.

The client will use it to get metadata. Server will use it to send metadata.